astropy/nddata/nduncertainty.py

```diff
@@ -64,7 +75,7 @@ def parent_nddata(self):
     def parent_nddata(self, value):
         self._parent_nddata = value
 
-    @property
+    @abc.abstractproperty
```

Comment:
As far as I understand any `abstractproperty`  can be implemeneted as class attribute. This way we just make sure that any subclass has something like that.